caddyconfig/caddyfile/lexer.go

```diff
@@ -78,16 +79,17 @@ func (l *lexer) next() bool {
 	var val []rune
 	var comment, quoted, btQuoted, escaped bool
 
-	makeToken := func() bool {
+	makeToken := func(quoted rune) bool {
 		l.token.Text = string(val)
+		l.token.wasQuoted = quoted
 		return true
 	}
 
 	for {
 		ch, _, err := l.reader.ReadRune()
 		if err != nil {
 			if len(val) > 0 {
-				return makeToken()
+				return makeToken(-1)
```

Comment:
This could probably be 0, right? No quote character is going to be delimited as 0x0.

Reply:
I guess? I just googled to see what the right value is for "not a rune" and a StackOverflow post had -1 🤷‍♂️ I don't know what's best but I'm indifferent, can change if you prefer

Reply:
Technically 0 is a valid character (null byte character), but it will never be a valid quotation mark or string encapsulator in the Caddyfile. So for convenience we can use the default zero-value (0) instead of -1 if that's easier.